File: socialgo/src/main/java/com/fungo/socialgo/share/weibo/WBShareCallbackActivity.java
package com.fungo.socialgo.share.weibo;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import com.fungo.socialgo.share.config.PlatformConfig;
import com.fungo.socialgo.share.config.PlatformType;
import com.fungo.socialgo.share.SocialApi;
import com.sina.weibo.sdk.constant.WBConstants;
import com.sina.weibo.sdk.share.WbShareCallback;


/**
 * Created by tsy on 16/8/4.
 */
public class WBShareCallbackActivity extends Activity implements WbShareCallback {

    protected SinaWBHandler mSinaWBHandler = null;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SocialApi api = SocialApi.get(this.getApplicationContext());
        this.mSinaWBHandler = (SinaWBHandler) api.getSSOHandler(PlatformType.SINA_WB);
        this.mSinaWBHandler.onCreate(this, PlatformConfig.getPlatformConfig(PlatformType.SINA_WB));

        if (this.getIntent() != null) {
            this.handleIntent(this.getIntent());
        }
    }

    protected final void onNewIntent(Intent paramIntent) {
        super.onNewIntent(paramIntent);
        SocialApi api = SocialApi.get(this.getApplicationContext());
        this.mSinaWBHandler = (SinaWBHandler) api.getSSOHandler(PlatformType.SINA_WB);
        this.mSinaWBHandler.onCreate(this, PlatformConfig.getPlatformConfig(PlatformType.SINA_WB));

        this.handleIntent(this.getIntent());
    }

    protected void handleIntent(Intent intent) {
        this.mSinaWBHandler.onNewIntent(intent, this);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        this.mSinaWBHandler.onActivityResult(requestCode, resultCode, data);
    }

    @Override
    public void onWbShareSuccess() {
        this.mSinaWBHandler.onResponse(WBConstants.ErrorCode.ERR_OK, "");
        finish();
    }

    @Override
    public void onWbShareCancel() {
        this.mSinaWBHandler.onResponse(WBConstants.ErrorCode.ERR_CANCEL, "");
        finish();
    }

    @Override
    public void onWbShareFail() {
        this.mSinaWBHandler.onResponse(WBConstants.ErrorCode.ERR_FAIL, "onWbShareFail");
        finish();
    }

    @Override
    public void onPointerCaptureChanged(boolean hasCapture) {

    }
}

File: socialgo/src/main/java/com/fungo/socialgo/share/weibo/WBShareCallbackActivity.kt
package com.fungo.socialgo.share.weibo

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import com.fungo.socialgo.share.SocialApi
import com.fungo.socialgo.share.config.PlatformConfig
import com.fungo.socialgo.share.config.PlatformType
import com.sina.weibo.sdk.constant.WBConstants
import com.sina.weibo.sdk.share.WbShareCallback

/**
 * @author Pinger
 * @since 18-7-20 下午3:53
 * 微博分享后的回调
 */

class WBShareCallbackActivity : Activity(), WbShareCallback {

    private var mSinaWBHandler: SinaWBHandler? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val api = SocialApi.get(this.applicationContext)
        this.mSinaWBHandler = api.getSSOHandler(PlatformType.SINA_WB) as SinaWBHandler
        this.mSinaWBHandler!!.onCreate(this, PlatformConfig.getPlatformConfig(PlatformType.SINA_WB))

        if (this.intent != null) {
            this.handleIntent(this.intent)
        }
    }


    override fun onNewIntent(paramIntent: Intent) {
        super.onNewIntent(paramIntent)
        val api = SocialApi.get(this.applicationContext)
        this.mSinaWBHandler = api.getSSOHandler(PlatformType.SINA_WB) as SinaWBHandler
        this.mSinaWBHandler!!.onCreate(this, PlatformConfig.getPlatformConfig(PlatformType.SINA_WB))

        this.handleIntent(this.intent)
    }

    private fun handleIntent(intent: Intent) {
        this.mSinaWBHandler?.onNewIntent(intent, this)
    }


    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        super.onActivityResult(requestCode, resultCode, data)
        this.mSinaWBHandler?.onActivityResult(requestCode, resultCode, data)
    }


    override fun onWbShareFail() {
        this.mSinaWBHandler?.onResponse(WBConstants.ErrorCode.ERR_FAIL, "onWbShareFail")
        finish()
    }

    override fun onWbShareCancel() {
        this.mSinaWBHandler?.onResponse(WBConstants.ErrorCode.ERR_CANCEL, "")
        finish()
    }

    override fun onWbShareSuccess() {
        this.mSinaWBHandler?.onResponse(WBConstants.ErrorCode.ERR_OK, "")
        finish()
    }
}